nodkz/mongodb-memory-server
# Fix/improveCode

- add enum for all resolveConfig Variables
- remove alias "reInitializePackageJson"
- remove value "LATEST_VERSION"
- set default version inside resolveConfig
- test(MongoBinaryDownload): replace "process.env.variable" with "process.env[values.variable]"
- refactor "should use direct download"
- fix multiline test name
- refactor "should skip download if binary tar exists"
- refactor "should pick up proxy from env vars"
- refactor "should not reject unauthorized"
- refactor "should reject unauthorized"
- refactor "makeMD5check returns true"
- refactor "makeMD5check throws an error"
- refactor "disables makeMD5check validation"
- makeMD5check: add more tsdoc
- test(MongoBinary): replace "process.env.variable" with "process.env[values.variable]"
- test(MongoBinary): refactor tests
- test(MongoInstance): refactor to use "isAlive"
- test(resolveConfig): use "fs.promises" instead of "promisify"
- test(replset-single): add description on why to use "ts-expect-error"
- test(replset-multi): directly use db "admin"
- test(MongoMemoryServer): add description on why to use "ts-expect-error"
- refactor(MongoBinary): replace "promisify" with "fs.promises"
- refactor(MongoBinaryDownload): replace "promisify" with "fs.promises"
- refactor(getos): replace "promisify" with "fs.promises"
- add function "pathExists"
- remove function "locationExists"
- test that "statPath" throws an error if code is not "ENOENT"
- test that assertion throws default error if none provided
- assertion: remove error code from default error
- refactor(MongoBinaryDownload): replace "existsSync" with "utils.pathExists"
- add more tsdoc to "statPath" & "pathExists"
- remove function "getCachePath" (because it just an alias to `cache.get`, which is public anyway)
- getSystemPath: also check for execute permission
- feat(MongoInstance): run: add check that the mongoBinary has sufficient permissions
- feat(MongoBinaryDownload): startDownload: add check that the downloadDir has sufficient permissions
- eslintrc
  - add environment "node"
  - set rule "no-unused-expressions" to warn
  - comma-dangle: change "functions" to "never"
  - set rule "no-else-return" to "warn"
  - add rule "padding-line-between-statements" for "return"
  - add rule "padding-line-between-statements" for "if"
  - add rule "padding-line-between-statements" for "function" & "class"
  - add rule "padding-line-between-statements" for "import"

---

[here](https://unix.stackexchange.com/a/317446/325185) is an good explanation on which permissions are needed for an directory for this package
